src/components/Dropzone/Dropzone.tsx

```diff
@@ -0,0 +1,214 @@
+import React, { useState, ChangeEvent, DragEvent, useRef } from 'react'
+import styled from '@emotion/styled'
+import { css } from '@emotion/react'
+import {
+  color,
+  space,
+  radius,
+  transition,
+  fontWeight,
+  device,
+} from '../../theme'
+import { H4 } from '../Heading'
+import { SmallText } from '../Text'
+import { BaseButton } from '../BaseButton'
+import ILLUSTRATION from './files.svg'
+import { Button } from '../Button'
+
+interface DropOverlayProps {
+  isDragging: boolean
+}
+
+export interface DropzoneProps {
+  title: string
+  subtitle: string
+  action: string
+  mobileAction?: string
+  dropTitle: string
+  accept?: string[]
+  multiple?: boolean
+  ariaLabel?: string
+  onFileChange: (files: FileList | File) => void
+  onUnacceptedFileChange?: () => void
+}
+
+const DropOverlay = styled.div<DropOverlayProps>`
+  position: fixed;
+  inset: 0;
+  background-color: ${color.actionBackground};
+  pointer-events: none;
+  z-index: 9999;
+  opacity: 0;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  transition: ${transition};
+  h2 {
+    margin-block: 0;
+    margin-inline: 0;
+  }
+  ${({ isDragging }) =>
+    isDragging &&
+    css`
+      opacity: 1;
+      pointer-events: auto;
+    `}
+`
+
+const DropArea = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  width: 100%;
+  border-radius: ${radius.lg};
+  padding-block-start: ${space[16]};
+  padding-block-end: ${space[24]};
+  padding-inline: ${space[16]};
+  text-align: center;
+  position: relative;
+  background-color: ${color.elevatedBackground};
+
+  @media ${device.tablet} {
+    background-color: ${color.actionBackground};
+    border: 2px dashed ${color.earth};
+  }
+`
+
+const Illustration = styled.img`
+  cursor: pointer;
+`
+
+const Title = styled(H4)`
```

Comment:
This is now always an h4. Would it be an idea to accept a prop for the `as` attribute so the user of the Dropzone component can configure how this heading should be interpreted? Maybe an h4 doesn't make sense in all situations.

Reply:
Looking at other components that aren't specifically made for holding text (e.g. `Card`) I think it's okay to make assumptions about the hierarchy on the page. But let me know if you think otherwise @TicketSwap/frontend I'm also okay with adding the prop.